base/applications/rapps/installed.cpp

```diff
@@ -12,149 +12,226 @@
-                                          (LPBYTE) szParentKeyName.GetBuffer(MAX_PATH),
-                                          &dwSize) == ERROR_SUCCESS);
-            szParentKeyName.ReleaseBuffer();
-
-            dwType = REG_SZ;
-            dwSize = MAX_PATH * sizeof(WCHAR);
-            if (RegQueryValueExW(Info.hSubKey,
-                                 L"DisplayName",
-                                 NULL,
-                                 &dwType,
-                                 (LPBYTE) szDisplayName.GetBuffer(MAX_PATH),
-                                 &dwSize) == ERROR_SUCCESS)
+            ItemIndex++;
+
+            szKeyName.ReleaseBuffer();
+            if (RegOpenKeyW(hKey, szKeyName.GetString(), &hSubKey) == ERROR_SUCCESS)
             {
-                szDisplayName.ReleaseBuffer();
-                if (EnumType < ENUM_ALL_INSTALLED || EnumType > ENUM_UPDATES)
-                    EnumType = ENUM_ALL_INSTALLED;
+                BOOL bSuccess = FALSE;
+                CInstalledApplicationInfo *Info = new CInstalledApplicationInfo(RootKeyEnum[i] == HKEY_CURRENT_USER, hSubKey);
+                Info->szKeyName = szKeyName;
 
-                if (!bIsSystemComponent)
+                // check for failure. if failed to init, Info->hSubKey will be set to NULL
+                if (Info->hSubKey)
                 {
-                    if ((EnumType == ENUM_ALL_INSTALLED) || /* All components */
-                        ((EnumType == ENUM_INSTALLED_APPLICATIONS) && (!bIsUpdate)) || /* Applications only */
-                        ((EnumType == ENUM_UPDATES) && (bIsUpdate))) /* Updates only */
+                    // those items without display name are ignored
+                    if (Info->GetApplicationString(L"DisplayName", Info->szDisplayName))
                     {
-                        if (!lpEnumProc(ItemIndex, szDisplayName, &Info, param))
-                            break;
+                        Info->GetApplicationString(L"DisplayVersion", Info->szDisplayVersion);
+                        Info->GetApplicationString(L"Publisher", Info->szPublisher);
+                        Info->GetApplicationString(L"RegOwner", Info->szRegOwner);
+                        Info->GetApplicationString(L"ProductID", Info->szProductID);
+                        Info->GetApplicationString(L"HelpLink", Info->szHelpLink);
+                        Info->GetApplicationString(L"HelpTelephone", Info->szHelpTelephone);
+                        Info->GetApplicationString(L"Readme", Info->szReadme);
+                        Info->GetApplicationString(L"Contact", Info->szContact);
+                        Info->GetApplicationString(L"URLUpdateInfo", Info->szURLUpdateInfo);
+                        Info->GetApplicationString(L"URLInfoAbout", Info->szURLInfoAbout);
+                        Info->GetApplicationString(L"Comments", Info->szComments);
+                        Info->GetApplicationString(L"InstallDate", Info->szInstallDate);
```

Comment:
In the registry it's possible that this is not a string but a DWORD.

For example, `xampp` has a DWORD here:
`1546720283`, which (according to the control panel) translates to `08-dec-19`

Reply:
hmmm, a UNIX-timestamp?